src/modules/job-manager/alloc.c

```diff
@@ -240,6 +240,7 @@ static void alloc_response_cb (flux_t *h, flux_msg_handler_t *mh,
             if (event_job_post_pack (ctx->event, job, "alloc", NULL) < 0)
                 goto teardown;
         }
+        job->alloc_success = 1;
```

Comment:
Can `job->has_resources` work here, instead of adding another flag?

Reply:
ahh yes, its sent in `event_job_update()` so just didn't see it.